Black box rule rewrite (close #209) (#212)
Benjamin-Lee/deep-rules: content/10.blackbox.md

## Tip 8: Your DL models can be more transparent {#blackbox}

> In ML, a model's interpretability refers to the ability to identify the discriminative features that influence or sway the predictions.

Reviewer: ```suggestion
Model interpretability is a broad concept.
In certain cases, the goal behind interpretation is to understand the underlying data generating processes while in other cases the goal is to understand why a model made the prediction that it did for a specific example or set of examples.
In much of the ML literature, including in our guidelines,  the concept of model interpretability refers to the ability to identify the discriminative features that influence or sway the predictions.
```

> ML models vary widely in terms of interpretability, ranging from being fully transparent to being a “black-box”.

Reviewer: ```suggestion
ML models vary widely in terms of interpretability: some are fully transparent while others are considered to be "black-boxes" that make predictions with little ability to examine why.
```

> In the former case, a clear description of the features' importance can found, such as with logistic regression or decision trees. 

Reviewer: ```suggestion
Logistic regression and decision tree models are generally considered interpretable, while deep neural networks are often considered among the most difficult to interpret because they can have many parameters and non-linear relationships. 
```

> However, DL models are often considered to be among the most difficult to interpret because they have many parameters and potentially non-linear relationships between their features.

Reviewer: ```suggestion
```

> Model interpretability is particularly important in biomedicine, where subsequent decision making requires human input.

Reviewer: ```suggestion
Model interpretability is particularly important in biomedicine, where subsequent decision making often requires human input.
```

> Many authors attribute the lack of pervasiveness of DL tools in healthcare because of the inability to understand what these models learn [@doi:10.1109/JBHI.2016.2636665; @doi:10.1038/s41551-018-0315-x]. 

Reviewer: ```suggestion
Many authors attribute a lower uptake of DL tools in healthcare to interpretability challenges [@doi:10.1109/JBHI.2016.2636665; @doi:10.1038/s41551-018-0315-x]. 
```

> Luckily, strategies to interpret both ML and DL models are rapidly emerging and the literature on the topic is currently growing at an exponential rate [@arxiv:2001.02522].

Reviewer: ```suggestion
Strategies to interpret both ML and DL models are rapidly emerging, and the literature on the topic is growing at an exponential rate [@arxiv:2001.02522].
```

ML models vary widely in terms of interpretability: some are fully transparent while others are considered to be "black-boxes" that make predictions with little ability to examine why.
Logistic regression and decision tree models are generally considered interpretable, while deep neural networks are often considered among the most difficult to interpret because they can have many parameters and non-linear relationships. 

Model interpretability is particularly important in biomedicine, where subsequent decision making often requires human input.
Many authors attribute a lower uptake of DL tools in healthcare to interpretability challenges [@doi:10.1109/JBHI.2016.2636665; @doi:10.1038/s41551-018-0315-x]. 
Strategies to interpret both ML and DL models are rapidly emerging, and the literature on the topic is growing at an exponential rate [@arxiv:2001.02522].
Therefore, instead of recommending specific methods for either DL-specific or general-purpose model interpretation, we suggest consulting [@url:https://christophm.github.io/interpretable-ml-book/] which is freely available and continually updated.

> In conclusion, analyze your DL models to identify how their inputs are interacting as they generate their predictions, especially when your models are being used for critical decision making.

Reviewer: ```suggestion
Model interpretation is an open, active area of research.
It is becoming more feasible to interpret models with many parameters and non-linear relationships, but in many cases simpler models remain substantially easier to interpret than more complex ones.
When deciding on a machine learning approach and model architecture, consider an interpretability versus accuracy tradeoff.
A challenge in considering this tradeoff is that the extent to which one trades interpretability for accuracy depends on the problem itself.
When the features provided to the model are already highly relevant to the task at hand, a simpler, interpretable model that gives up only a little performance when compared to a very complex one more useful in many settings.
On the other hand, if features must be combined in complex ways to be meaningful for the task, the performance difference of a model capable of capturing that structure may outweigh the interpretability costs.
An appropriate choice can only be made after careful consideration, which often includes estimating the performance of a simple, linear model that serves as a baseline.
```

## Tip 8: Your DL models can be more transparent {#blackbox}

> In ML, a model's interpretability refers to the ability to identify the discriminative features used for learning.

Reviewer 2: > features used for learning.

and influence or sway the predictions?

> In ML, a model's interpretability refers to the ability to identify the discriminative features used for learning and that influence or sway the predictions.

Reviewer: ```suggestion
In ML, a model's interpretability refers to the ability to identify the discriminative features that influence or sway the predictions.
```
Does it make sense to further constrain this to _within the space of the predictions_? For models with nonlinearities - in some ranges of input a feature might greatly influence results, but that may be irrelevant with the space of inputs used for predictions.

Author: >that may be irrelevant with the space of inputs used for predictions.
Yes, I think your suggestion is fair, although I do think it can be interesting to see how the model fares on inputs far from the expected input space.

> ML models vary in widely in terms of interpretability, ranging from being fully transparent to being a “black-box”.

Reviewer 2: > ML models vary in widely in terms of interpretability, ranging from being fully transparent to being a “black-box”.

ML models vary ~~in~~ widely in terms of interpretability, ranging from being fully transparent to being a “black-box**.”**

In the former case, a clear description of the features' importance can found, such as with logistic regression or decision trees. 

> However, DL models are harder to interpret when compared to other ML models because of their large number of parameters and the non-linear relationships between their features.

Reviewer 2: > harder to interpret when compared to

harder to interpret ~~when~~ compared to

> However, DL models are harder to interpret compared to other ML models because of their large number of parameters and the non-linear relationships between their features.

Reviewer: ```suggestion
However, DL models are often considered to be among the most difficult to interpret because they have many parameters and potentially non-linear relationships between their features.
```

> Model interpretability is notably important in the biomedical field, where the decisions of models can mean the difference between life and death.

Reviewer: ```suggestion
Model interpretability is particularly important in biomedicine, where subsequent decision making requires human input.
```

Many authors attribute the lack of pervasiveness of DL tools in healthcare because of the inability to understand what these models learn [@doi:10.1109/JBHI.2016.2636665; @doi:10.1038/s41551-018-0315-x]. 

> Luckily, there are many strategies to interpret both ML and DL models.

Author: I'm personally not wedded to the use of the word "many". However, the Interpretable ML book describes ten model agnostic methods (including SHAP and LIME) and has four more sections in progress specifically for ML. How about a compromise such as: **Luckily, there are several strategies to interpret DL models**

> Luckily, there are many strategies to interpret both ML and DL models.

Reviewer: ```suggestion
Luckily, strategies to interpret both ML and DL models are rapidly emerging and the literature on the topic is currently growing at an exponential rate [@arxiv:2001.02522].
```

> In fact, due to the increasing demand for interpretable models, the literature on the topic is currently growing at an exponential rate [@arxiv:2001.02522].

Reviewer: ```suggestion
```

> Therefore, instead of recommending specific methods, we suggest consulting a recent overview.

Reviewer 3: I understand the point, there are too many new method, so any list we give would be outdated. Fell free to reject my previous comment

> Therefore, instead of recommending specific methods for either DL-specific or general-purpose model interpretation, we suggest consulting a recent overview.

Reviewer: ```suggestion
Therefore, instead of recommending specific methods for either DL-specific or general-purpose model interpretation, we suggest consulting [@url:https://christophm.github.io/interpretable-ml-book/] which is freely available and continually updated.
```

> We particularly recommend [@url:https://christophm.github.io/interpretable-ml-book/] which is freely available and continually updated.

Reviewer: ```suggestion
```

> In conclusion, when using DL, be sure to analyze your model to identify how its inputs are interacting as it generates its predictions.

Reviewer 3: Please consider expanding on the final remark:
**In conclusion, model transparency may be critical for DL projects, so consider ML models that support feature discrimination or post hoc techniques to understand the learned features. With you will be able to analyze your model to identify how its inputs are interacting as it generates its predictions.**

Author: I'm trying to understand the first sentence, so could you help me understand? 

We now say earlier in the tip that there are model-agnostic and DL-specific methods for interpretation. So, when you say "so consider ML models" do you mean that the reader should consider ML and not DL models that are especially interpretable (eg decision trees)? Or is that a recommendation for which types of baseline models they should use (per tip 2)?

Sorry for my confusion

Reviewer 3: I agree this sentence was not clear in its context.
we could simplify it with: **(...) so consider using approaches for feature discrimination to understand the learned features.**

> In conclusion, analyze your DL models to identify how their inputs are interacting as they generate their predictions, especially when your models are being used for critical decision making.

Reviewer: This is not really addressed in this section anymore. Before it was suggested to train both simple and complex models, which is not a bad suggestion in many cases.

Author: Sorry, can you clarify your comment a little bit? Are you suggesting adding back our suggestion to train a parallel simple model for the purposes of interpretation (in addition to baselining)?

Reviewer: This sentence says that the conclusion is to analyze deep learning models to identify how their inputs are interacting. Our paragraph above mostly says these models are considered among the most difficult to interpret but new techniques are emerging.

I don't know if we want to change our conclusion to be about the paragraph that's here, or if we want to revise our guidelines to provide advice/examples around why it's so important to analyze and interpret models (which would probably require discussion of things like https://medium.com/@jrzech/what-are-radiological-deep-learning-models-actually-learning-f97a546c5b98 from @jrzech
